src/metabase/search/filter.clj

```diff
@@ -13,149 +13,73 @@
-;;                                         Required Filters                                         ;
-;; ------------------------------------------------------------------------------------------------;;
-
-(defmulti ^:private archived-clause
-  "Clause to filter by the archived status of the entity."
-  {:arglists '([model archived?])}
-  (fn [model _] model))
-
-(defmethod archived-clause :default
-  [model archived?]
-  [:= (search.config/column-with-model-alias model :archived) archived?])
-
-;; Databases can't be archived
-(defmethod archived-clause "database"
-  [_model archived?]
-  (if archived?
-    false-clause
-    true-clause))
-
-(defmethod archived-clause "indexed-entity"
-  [_model archived?]
-  (if-not archived?
-    true-clause
-    false-clause))
-
-;; Table has an `:active` flag, but no `:archived` flag; never return inactive Tables
-(defmethod archived-clause "table"
-  [model archived?]
-  (if archived?
-    false-clause ; No tables should appear in archive searches
-    [:and
-     [:= (search.config/column-with-model-alias model :active) true]
-     [:= (search.config/column-with-model-alias model :visibility_type) nil]]))
-
-(defn- sandboxed-or-impersonated-user? []
-  ;; TODO FIXME -- search actually currently still requires [[metabase.api.common/*current-user*]] to be bound,
-  ;; because [[metabase.public-settings.premium-features/sandboxed-or-impersonated-user?]] requires it to be bound.
-  ;; Since it's part of the search context it would be nice if we could run search without having to bind that stuff at
-  ;; all.
-  (assert @@(requiring-resolve 'metabase.api.common/*current-user*)
-          "metabase.api.common/*current-user* must be bound in order to use search for an indexed entity")
-  (premium-features/sandboxed-or-impersonated-user?))
-
-(mu/defn ^:private search-string-clause-for-model
-  [model                :- SearchableModel
-   search-context       :- SearchContext
-   search-native-query? :- [:maybe :boolean]]
-  (when-let [query (:search-string search-context)]
-    (into
-     [:or]
-     (for [column           (cond->> (search.config/searchable-columns-for-model model)
-                              (not search-native-query?)
-                              (remove #{:dataset_query})
-
-                              true
-                              (map #(search.config/column-with-model-alias model %)))
-           wildcarded-token (->> (search.util/normalize query)
-                                 search.util/tokenize
-                                 (map search.util/wildcard-match))]
-       (cond
-         (and (= model "indexed-entity") (sandboxed-or-impersonated-user?))
-         [:= 0 1]
-
-         (and (#{"card" "dataset"} model) (= column (search.config/column-with-model-alias model :dataset_query)))
-         [:and
-          [:= (search.config/column-with-model-alias model :query_type) "native"]
```

Comment:
Filtering for native queries is implemented here https://github.com/metabase/metabase/blob/188649a8a757e4830c029676e45437bfac1b71c7/src/metabase/search/impl.clj#L255-L256